core/controllers/base_test.py

```diff
@@ -712,6 +770,34 @@ def test_controller_class_names(self):
         self.assertGreater(num_handlers_checked, 150)
 
 
+class HasSeenTutorialTests(test_utils.GenericTestBase):
+
+    class MockHandler(base.BaseHandler):
+        def get(self):
+            assert self.has_seen_editor_tutorial is True
```

Comment:
Don't use asserts, we should test exceptions instead.

Reply:
Here we are checking if `has_seen_editor_tutorial` becomes true after `user_services.record_user_started_state_editor_tutorial()` so assert is required. Another approach can be to render_json a dict containing values of has_seen_editor_tutorial and has_seen_translation_tutorial and doing a self.assert in the test function. That way we can do the pre and post checks. What do you think?

Reply:
I don't understand why assert is required?

What's wrong with:

```python
if not self.has_seen_editor_tutorial:
    raise Exception('useful description')
if not self.has_seen_translation_tutorial:
    raise Exception('useful description')
self.render_json({})
```

Reply:
This is the test file(base_test) and I think using asserts in our test files was the normal convention? I will look after the other changes that you mentioned but I think we need assert to check if the values are getting updated

Reply:
Removed the assert statements from base.py. PTAL